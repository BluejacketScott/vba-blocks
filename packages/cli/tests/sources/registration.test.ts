import {
  getRegistrationId,
  getRegistrationSource,
  fromSnapshot
} from '../../src/sources/registration';

const snapshot = {
  name: 'a',
  version: '1.0.0',
  features: [],
  dependencies: []
};

test('should get registration id from snapshot', () => {
  const id = getRegistrationId(snapshot);
  expect(id).toEqual('a@1.0.0');
});

test('should get registration id from name + version', () => {
  const id = getRegistrationId('a', '1.0.0');
  expect(id).toEqual('a@1.0.0');
});

test('should get registration source', () => {
  let source = getRegistrationSource('type', 'value');
  expect(source).toEqual('type+value');

  source = getRegistrationSource('type', 'value', 'details');
  expect(source).toEqual('type+value#details');
});

test('should get registration from snapshot', () => {
  const registration = fromSnapshot(snapshot, 'type+value#details');
  expect(registration).toMatchSnapshot();
});
