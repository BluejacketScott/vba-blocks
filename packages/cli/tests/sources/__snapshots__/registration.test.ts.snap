// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get registration from snapshot 1`] = `
Object {
  "dependencies": Array [],
  "features": Array [],
  "id": "a@1.0.0",
  "name": "a",
  "source": "type+value#details",
  "version": "1.0.0",
}
`;
